### ISSUE TITLE
Port twisted.python.reflect.objgrep to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6986]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)| @multani reported|
|-|-|
|Trac ID|trac#6986|
|Type|enhancement|
|Created|2014-02-25 16:04:13Z|

While porting `twisted.python.reflect` in [#5929](https://github.com/twisted/twisted/issues/5929), some odd cases were discovered while porting the `objgrep()` function. See the following comments for reference:

 * [[[[#5929](https://github.com/twisted/twisted/issues/5929)](https://github.com/twisted/twisted/issues/5929)](https://github.com/twisted/twisted/issues/5929)](https://github.com/twisted/twisted/issues/5929)#comment:40
 * #5929#comment:41
 * #5929#comment:43
 * #5929#comment:44

Although `objgrep()` isn't use *inside* Twisted itself, it's blocking the porting of the whole `twisted.python.reflect` module. So it has been decided to do *most* of the porting of the module in [#5929](https://github.com/twisted/twisted/issues/5929) but disable `objgrep()` for Python 3.

The goal of this ticket is to finish to port the whole module by handling the use-cases explained in the comment in [[#5929](https://github.com/twisted/twisted/issues/5929)](https://github.com/twisted/twisted/issues/5929).

This ticket has to be done only once #5929 has been fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__6986 6986
type__enhancement enhancement
reporter__multani multani
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1393344253000000 1393344253000000
changetime__1594426734250477 1594426734250477
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_1"></a>|@wallrj commented|
|-|-|

(In [41754]) Merge python3-reflect-5929-2

Author: multani
Reviewers: radix, thijs, rwall, glyph
Fixes: [#5929](https://github.com/twisted/twisted/issues/5929)
Refs: [#6239](https://github.com/twisted/twisted/issues/6239), [[#6986](https://github.com/twisted/twisted/issues/6986)](https://github.com/twisted/twisted/issues/6986), [[#6859](https://github.com/twisted/twisted/issues/6859)](https://github.com/twisted/twisted/issues/6859)

Port most of twisted.python.reflect to Python3.

Porting objgrep posed some problems and has been moved to a separate ticket #6986.

Porting getcurrent and isinst also posed problems and it was decided to deprecate them in  #6859.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

https://github.com/twisted/twisted/pull/1347

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @glyph|
|-|-|

Approved, with some minor review feedback: https://github.com/twisted/twisted/pull/1347#pullrequestreview-446731872

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set status to `closed`|
|-|-|

Oops, [#6896](https://github.com/twisted/twisted/issues/6896)#comment:1 should have closed this one instead.
